network/handlers/drain_test.go

```diff
@@ -55,41 +109,56 @@ func TestDrainMechanics(t *testing.T) {
 		t.Errorf("Probe status = %d, wanted %d", got, want)
 	}
 
-	// Start to drain, and cancel the context when it returns.
+	// Start to drain, and close the channel when it returns.
 	done := make(chan struct{})
 	go func() {
 		defer close(done)
 		drainer.Drain()
 	}()
 
 	select {
-	case <-time.After(40 * time.Millisecond):
-		// Drain is blocking.
 	case <-done:
 		t.Error("Drain terminated prematurely.")
+	case <-init:
+		// OK.
 	}
+	mt.advance(40 * time.Millisecond)
+
 	// Now send a request to reset things.
+	rc := mt.resetCalls
 	drainer.ServeHTTP(w, req)
+	if mt.resetCalls != rc+1 {
+		t.Errorf("ResetCalls = %d, want: %d", mt.resetCalls, rc+1)
+	}
 
 	// Check for 503 as a probe response when shutting down.
 	resp = httptest.NewRecorder()
 	drainer.ServeHTTP(resp, probe)
 	if got, want := resp.Code, http.StatusServiceUnavailable; got != want {
 		t.Errorf("Probe status = %d, wanted %d", got, want)
 	}
+	// Verify no reset was called.
+	if mt.resetCalls != rc+1 {
+		t.Errorf("ResetCalls = %d, want: %d", mt.resetCalls, rc+1)
+	}
 
 	for i := 0; i < 3; i++ {
+		mt.advance(40 * time.Millisecond)
 		select {
-		case <-time.After(40 * time.Millisecond):
-			// Drain is blocking.
 		case <-done:
 			t.Error("Drain terminated prematurely.")
+		default:
+			// OK
 		}
 		// For the last one we don't want to reset the drain timer.
 		if i < 2 {
 			drainer.ServeHTTP(w, req)
 		}
 	}
+	if mt.resetCalls != rc+3 {
```

Comment:
If we're going to just use `+n`, why not start at zero and just say `n`?  We are partially assuming total knowledge :)

Reply:
Natural progress of things 😊 ?